packages/spezi-firebase-fhir/src/resources/evidenceVariable.ts

```diff
@@ -0,0 +1,159 @@
+            range: rangeSchema.optional(),
+            note: annotationSchema.array().optional(),
+          }),
+        )
+        .optional(),
+      groupMeasure: z
+        .lazy(() =>
+          codeSchema.pipe(
+            z.enum([
+              'mean',
+              'median',
+              'mean-of-mean',
+              'mean-of-median',
+              'median-of-mean',
+              'median-of-median',
+            ]),
+          ),
+        )
+        .optional(),
+      _groupMeasure: elementSchema.optional(),
+    }),
+  )
+
+export const untypedEvidenceVariableSchema = z.lazy(() =>
+  domainResourceSchema.extend({
+    resourceType: z.literal('EvidenceVariable').readonly(),
+    url: uriSchema.optional(),
+    _url: elementSchema.optional(),
+    identifier: identifierSchema.array().optional(),
+    version: stringSchema.optional(),
+    _version: elementSchema.optional(),
+    name: stringSchema.optional(),
+    _name: elementSchema.optional(),
+    title: stringSchema.optional(),
+    _title: elementSchema.optional(),
+    shortTitle: stringSchema.optional(),
+    _shortTitle: elementSchema.optional(),
+    subtitle: stringSchema.optional(),
+    _subtitle: elementSchema.optional(),
+    status: publicationStatusSchema,
+    _status: elementSchema.optional(),
+    date: dateTimeSchema.optional(),
+    _date: elementSchema.optional(),
+    publisher: stringSchema.optional(),
+    _publisher: elementSchema.optional(),
+    contact: contactDetailSchema.array().optional(),
+    description: markdownSchema.optional(),
+    _description: elementSchema.optional(),
+    note: annotationSchema.array().optional(),
+    useContext: usageContextSchema.array().optional(),
+    copyright: markdownSchema.optional(),
+    _copyright: elementSchema.optional(),
+    approvalDate: dateSchema.optional(),
+    _approvalDate: elementSchema.optional(),
+    lastReviewDate: dateSchema.optional(),
+    _lastReviewDate: elementSchema.optional(),
+    effectivePeriod: periodSchema.optional(),
+    author: contactDetailSchema.array().optional(),
+    editor: contactDetailSchema.array().optional(),
+    reviewer: contactDetailSchema.array().optional(),
+    endorser: contactDetailSchema.array().optional(),
+    relatedArtifact: relatedArtifactSchema.array().optional(),
+    actual: booleanSchema.optional(),
+    _actual: elementSchema.optional(),
+    characteristic: evidenceVariableCharacteristicSchema.array().optional(),
+    handling: evidenceVariableHandlingSchema.optional(),
+    _handling: elementSchema.optional(),
+    category: z
+      .lazy(() =>
+        backboneElementSchema.extend({
+          name: stringSchema.optional(),
+          _name: elementSchema.optional(),
+          valueCodeableConcept: codeableConceptSchema.optional(),
+          valueQuantity: quantitySchema.optional(),
+          valueRange: rangeSchema.optional(),
+        }),
+      )
+      .array()
+      .optional(),
+  }),
+) satisfies ZodType<EvidenceVariable>
```

Comment:
Make use of `z.lazy` only where necessary and extract valueSets into the respective directory